### PR TITLE
Feature/pose queries

### PR DESCRIPTION
**Pull request by lucaro**
This adds the relevant functionality for pose-based queries.

### CODE REVIEW

File: cineast-api/src/main/java/org/vitrivr/cineast/api/messages/query/TemporalQuery.java [hunk 1]
```diff
@@ -76,4 +77,8 @@ public TemporalQueryConfig getTemporalQueryConfig() {
   public MessageType getMessageType() {
     return MessageType.Q_TEMPORAL;
   }
+
+  public void setTimeDistances(List<Float> distances) {
```
Comment (silvanheller) [inline cineast-api/src/main/java/org/vitrivr/cineast/api/messages/query/TemporalQuery.java:81]: If you have merged vitrivr-ng correctly, this field is not required. It is present in the `TemporalQueryConfig` (it's possible that this is my bad). If you open a PR for the vitrivr-ng branch, I can check that.

File: cineast-api/src/main/java/org/vitrivr/cineast/api/rest/resolvers/FileSystemThumbnailResolver.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public ResolutionResult resolve(String segmentId) {
     File[] candidates = dir.listFiles(new FilenameFilter() {
       @Override
       public boolean accept(File dir, String name) {
-        return name.startsWith(segmentId);
+        return name.startsWith("shot" + split[1] + "_" + split[2]);
```
Comment (silvanheller) [inline cineast-api/src/main/java/org/vitrivr/cineast/api/rest/resolvers/FileSystemThumbnailResolver.java:38]: While I haven't used these resolvers in the past, I'm not sure why these changes are required. Could you add a small comment, so we can check if the documentation or the default config requires an update?

File: cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java [hunk 3]
```diff
@@ -143,8 +141,8 @@ public boolean createSegmentEntity() {
       this.cottontail.client.create(entity);
 
       /* Create indexes. */
-      this.createIndexWithoutSchemaName(MediaSegmentDescriptor.ENTITY, MediaSegmentDescriptor.FIELDNAMES[0], HASH_UQ, txId);
-      this.createIndexWithoutSchemaName(MediaSegmentDescriptor.ENTITY, MediaSegmentDescriptor.FIELDNAMES[1], IndexType.HASH, txId);
+      this.createIndex(entityName, MediaSegmentDescriptor.FIELDNAMES[0], IndexType.BTREE_UQ, txId);
```
Comment (silvanheller) [inline cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java:144]: Is `HASH` now simply `BTREE` or are there also implementation changes in Cottontail?

File: cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java [hunk 4]
```diff
@@ -358,20 +341,9 @@ public static Type mapAttributeType(AttributeDefinition.AttributeType type) {
...
-      this.cottontail.client.create(index);
-    }catch(StatusRuntimeException e){
-      if(e.getStatus().getCode()== Status.ALREADY_EXISTS.getCode()){
-        LOGGER.warn("Index {} was not created because it already exists", indexName);
-        return false;
-      }
-      throw e;
-    }
-    return true;
+    this.cottontail.client.create(index);
```
Comment (silvanheller) [inline cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java:347]: I don't understand the diff here, I think something went wrong while merging. The functionality to create an Index for the ID seems to have been deleted, and the handling for if an index already exists seems also to have been deleted. Is this a merge-error (if so, I can fix that) or is this intentional?

**Review by silvanheller — CHANGES_REQUESTED** [review]
There are some changes which require additional documentation, and some which might be merge errors. I think it would be good to open also a PR for vitrivr-ng to `dev`, which we can have a look at in parallel.

File: cineast-api/src/main/java/org/vitrivr/cineast/api/messages/query/TemporalQuery.java (hunk 1 above)
Comment (lucaro) [reply cineast-api/src/main/java/org/vitrivr/cineast/api/messages/query/TemporalQuery.java:81]: I didn't touch vitrivr-ng at all, just used the branch that has the pose query interface. Without this setter, it didn't work. That branch might not be completely current though, I'll have to check.

File: cineast-api/src/main/java/org/vitrivr/cineast/api/rest/resolvers/FileSystemThumbnailResolver.java (hunk 2 above)
Comment (lucaro) [reply cineast-api/src/main/java/org/vitrivr/cineast/api/rest/resolvers/FileSystemThumbnailResolver.java:38]: This might be able to be removed again. It was needed to be compatible with the format of the v3c dataset. Maybe something like this should be more easily configurable.

File: cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java (hunk 3 above)
Comment (lucaro) [reply cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java:144]: This is a cottontail change, xodus comes with new index types. Maybe @ppanopticon can elaborate?

File: cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java (hunk 4 above)
Comment (lucaro) [reply cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java:347]: This looks like a merge error to me as well, git apparently got confused with the change order when I resurrected the dev branch.

File: cineast-api/src/main/java/org/vitrivr/cineast/api/messages/query/TemporalQuery.java (hunk 1 above)
Comment (lucaro) [reply cineast-api/src/main/java/org/vitrivr/cineast/api/messages/query/TemporalQuery.java]: With the most recent version, it's no longer needed, so I removed it.

File: cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java (hunk 3 above)
Comment (ppanopticon) [reply cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailEntityCreator.java:144]: This is simply a rename. 

What used to be called HASH and HASH_UQ were Tree based indexes all along (even for MapDB). I just took the opportunity to reflect that in the naming of the different enumerations.

**Review by silvanheller — COMMENTED** [review]
Looks good to me. I'll have to play around with the UI & import to see if it works, but generally speaking this works for me.

**Review by ppanopticon — CHANGES_REQUESTED** [review]
Tests still fail because the branch requires 0.15.0-SNAPSHOT version of Cottontail DB to run, which is not available as Docker Image yet.

We can finalise this PR, once Cottontail DB has been released.

**Review by silvanheller — APPROVED** [review]
LGTM! We'll wait with merging until v0.15 of Cottontail is out, but this looks really cool.

**Review by ppanopticon — APPROVED** [review]
